.github/workflows/rust-publish-dry-run-v2.yml

```diff
@@ -76,6 +86,23 @@ jobs:
         rm Cargo.toml
         mv Cargo.toml.bak Cargo.toml
 
+    # If a list of crates weren't provided, prepare a list by asking
+    # cargo-workspaces for the order it would publish the crates in, and filter
+    # the list of returned crates by crates that are marked for publishing.
+    - name: Prepare List of Crates Ordered by Publish-Order
+      if: inputs.crates == ''
+      run: |
+        echo "CRATES=\"$( \
+          cargo workspaces exec --no-bail \
+            sh -c ' \
+              name=$(basename $(pwd)); \
+              cargo metadata --format-version 1 --no-deps \
+                | jq -r '"'"'.packages[] | select(.publish != []) | .name'"'"' \
```

Comment:
The `select(.publish != [])` means select any package that doesn't have the `publish` variable set to `[]`. It's counter-intuitive, but the publish field in the metadata has the value `[]` only when `publish = false` in the `Cargo.toml`. That's because in the metadata the `publish` field is a list of registeries it can be published to.